great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -37,14 +38,13 @@
     Datasource,
     TestConnectionError,
 )
+from great_expectations.util import maybe_import_module
 
-SQLALCHEMY_IMPORTED = False
-try:
-    import sqlalchemy
+sqlalchemy = maybe_import_module("sqlalchemy")
 
-    SQLALCHEMY_IMPORTED = True
-except ImportError:
-    pass
+if TYPE_CHECKING:
+    # sqlalchemy is a hard requirement for type checking
+    import sqlalchemy  # noqa: TCH004
```

Comment:
If this is going to be a recurring problem with the `maybe_import_module()` we might want to just disable this rule.
It's only ever "caught" these false positives where we have a try import and an `If TYPE_CHECKING` import.

Reply:
The strategy is different now, see https://github.com/great-expectations/great_expectations/pull/7274#issuecomment-1452701350.